File: Problemas/1097A - Gennady and a Card Game/1097A_Gennady_And_A_Card_Game.py
# 1097A - Gennady and a Card Game
# https://codeforces.com/problemset/problem/1097/A

# Input:
# 1) carta_mesa = Carta en la mesa
carta_mesa = input()

# Input:
# 2) carta_mano = Cartas en la mano
cartas_mano = map(str, input().split())

# Extrae los números y tipos de las cartas en tu mano
cartas_numeros = []
cartas_tipos = []
for carta in cartas_mano:
    cartas_numeros.append(carta[0])
    cartas_tipos.append(carta[1])

# Revisa si tienes alguna carta para bajar
if carta_mesa[0] in cartas_numeros or carta_mesa[1] in cartas_tipos:
    print("YES")
else:
    print("NO")

File: Problemas/1097A - Gennady and a Card Game/1097A_Gennady_And_A_Card_Game.cpp
// 1097A - Gennady and a Card Game
// https://codeforces.com/problemset/problem/1097/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) carta_mesa = Carta en la mesa
    string carta_mesa;
    cin >> carta_mesa;

    // Input:
    // 2) carta_mano = Cartas en la mano
    string cartas_mano[5];
    for(int indice_carta = 0; indice_carta < 5; indice_carta++) {
        cin >> cartas_mano[indice_carta];
    }

    // Compara los números y tipos de las cartas en tu mano con el de la mesa
    bool bajar_carta = false;
    for(int indice_carta = 0; indice_carta < 5; indice_carta++) {
        string carta_actual = cartas_mano[indice_carta];
        if(carta_actual[0] == carta_mesa[0]) {
            bajar_carta = true;
        }
        if(carta_actual[1] == carta_mesa[1]) {
            bajar_carta = true;
        }
    }

    // Si hay una coincidencia en una carta, entonces imprime "YES", de lo contrario, imprime "NO"
    if(bajar_carta) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}